src/complex.rs

```diff
@@ -287,6 +287,24 @@ impl<T: Clone + Float> Complex<T> {
     pub fn is_nan(self) -> bool {
         self.re.is_nan() || self.im.is_nan()
     }
+
+    /// Checks if the given complex number is infinite
+    #[inline]
+    pub fn is_infinite(self) -> bool {
+        !self.is_nan() && (self.re.is_infinite() || self.im.is_infinite())
```

Comment:
As i said in the other PR, I don't think Wolfram's [`ComplexInfinity`](http://mathworld.wolfram.com/ComplexInfinity.html) is a useful definition for a general `is_infinite`. Yes, that's one concept of infinity, but they also have [`DirectedInfinity`](http://mathworld.wolfram.com/DirectedInfinity.html).  In polar form these are roughly "r e^i(undef)" and "∞ e^iθ" respectively.  We could also look at Maple's [`complex`](http://www.maplesoft.com/support/help/maple/view.aspx?path=complex) and [`cx_infinity`](http://www.maplesoft.com/support/help/maple/view.aspx?path=type%2finfinity) test, which requires both real and imaginary parts to be infinite.

But those are both high-level mathematical languages, not dealing with floating point properties.  For a more pragmatic programming view, we can look at Python's [`cmath`](https://docs.python.org/3/library/cmath.html) `isinf()`, which returns "True if either the real or the imaginary part of x is an infinity, and False otherwise."  The source is [here](https://github.com/python/cpython/blob/master/Modules/cmathmodule.c#L1111); it really is that simple.  I also looked for this function in standard C, C++, Haskell, and Ruby complex types, but didn't find it at all.